bfm/tests/CMakeLists.txt

```diff
@@ -15,25 +15,48 @@
 # You should have received a copy of the GNU Lesser General Public
 # License along with this library; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+cmake_minimum_required(VERSION 3.6)
+project(bfm_test C CXX)
+include(${BF_SCRIPTS_DIR}/cmake/flags/flags.cmake)
+include(CTest)
+enable_testing(true)
+enable_language(ASM_NASM)
 
-include(${CMAKE_INSTALL_PREFIX}/cmake/CMakeGlobal_Includes.txt)
+set(PROJECT_SRC_DIR ${CMAKE_CURRENT_LIST_DIR}/../src)
+set(PROJECT_TEST_DIR ${CMAKE_CURRENT_LIST_DIR}/../tests)
+set(PROJECT_INC_DIR ${CMAKE_CURRENT_LIST_DIR}/../include)
 
-# ------------------------------------------------------------------------------
-# Targets
-# ------------------------------------------------------------------------------
+include_directories(
+    ${PROJECT_INC_DIR}
+    ${BUILD_SYSROOT_TEST}/include
+    ${BUILD_SYSROOT_OS}/include
+)
 
-add_library(test_vmcall STATIC test_vmcall.cpp)
+link_directories(
+    ${BUILD_SYSROOT_TEST}/lib
+)
+
+set(BFM_TEST_CXX_FLAGS ${CMAKE_CXX_FLAGS} "-DENABLE_UNITTESTING=1")
+generate_flags(USERSPACE ADD_CXX_FLAGS ${BFM_TEST_CXX_FLAGS})
```

Comment:
I am assuming that generate_flags is a function that we added. Couple of notes:

- The above two lines takes the generic CMAKE_CXX_FLAGS and adds the BFM specific flags and then passes that to generate flags. Would it make more sense for generate flags to by default add CMAKE_CXX_FLAGS, and then all you would have to do is add the BFM specific flags, reducing this to a single line?
- Why are we manually adding ENABLE_UNITTESTING to BFM instead of by default adding this to all files when testing is enabled?
- Should the syntax for generate_flags look more like this based on the comments above:

```
generate_flags(
    TYPE userspace
    ADDITIONAL_CXX_FLAGS --yourmom
)
```

Also... notice I didn't use quotes. We should ensure that quotes are not used unless they need to be for everything because if you use quotes, sometimes CMake will pass those quotes to the command line which can cause issues.